### ISSUE TITLE
`pyreverse` does not show inheritance relation when using a flat folder

### ISSUE THREAD

**Issue by DudeNr33** [PyCQA/pylint#7686]
### Bug description

See [this SO question](https://stackoverflow.com/questions/74123298/pyreverse-not-connecting-classes-in-uml-class-diagram).

`Foo.py`, `Bar.py` and `__init__.py` are all located in the same folder.

`Foo.py`:
```python
class Foo():
    def print_foo(self):
        print("hi")
```

`Bar.py`:
```python
from Foo import Foo

class Bar(Foo):
    pass
```

`pyreverse` is executed from within the same folder.

### Command used

```shell
pyreverse -ASmy -o png .
```

### Pylint output

```shell
The resulting image shows both classes `Foo` and `Bar`, but not the inheritance relationship between them.
```

### Expected behavior

It could be argued that when using an `__init__.py` and therefore a package relative or absolute imports should be used, the code provided does run fine when just running e.g. `python -c "import Bar"` from inside the directory.

Therefore `pyreverse` should pick up the relationships between the two classes as well. 

### Pylint version

```shell
pylint 2.14.5
astroid 2.12.10
Python 3.10.5 (main, Jun  6 2022, 18:49:26) [GCC 12.1.0]
```

**Comment by nhadziosma1** [comment]
Are there any updates on this issue?

**Comment by atomic-light** [comment]
Same issue here now. Are there any solutions? I can look try to contribute, but I'm not a pro though.

**Comment by Pierre-Sassoulas** [comment]
Hey @atomic-light sure we can answer your question if you have some. Imo the first step is to add the test for that. Also do not hesitate to open a draft merge request early it's easier to talk about the issue when seeing the code directly.

**Comment by DudeNr33** [comment]
The way I would go at this would probably be:

1. Create a new functional test by creating a new folder beneath `tests/pyreverse/functional/class_diagrams` and check if the problem stated in the opening post or linked SO question can be reproduced
2. Debug the test to see what is going on - I could imagine that especially the `visit_importfrom` method could be worthwhile to investigate. https://github.com/pylint-dev/pylint/blob/3f93f1ee8b4544578be81b108f2f912e24d3f7ce/pylint/pyreverse/inspector.py#L240
